llm/codellama/README.md

```diff
@@ -0,0 +1,185 @@
+      "max_tokens": 256
+  }' | jq -r '.choices[0].text'
+```
+
+This returns the following completion:
+```python
+    if len(a) <= 1:
+        return a
+    pivot = a.pop(len(a)//2)
+    b = []
+    c = []
+    for i in a:
+        if i > pivot:
+            b.append(i)
+        else:
+            c.append(i)
+    b = quick_sort(b)
+    c = quick_sort(c)
+    res = []
+    res.extend(c)
+    res.append(pivot)
+    res.extend(b)
+    return res
+```
+
+## Scale up the service with SkyServe
+
+1. To scale up the Code Llama service is as simple as running the following command:
+```bash
+sky serve up -n code-llama ./endpoint.yaml
+```
+This will start the service with multiple replicas on cheapest available locations and accelerators, and SkyServe will automatically manage the replicas, monitor the healthiness of the replicas, and restart the replicas when needed.
+
+A single endpoint will be returned and any request sent to the endpoint will be routed to the healthy replicas.
+
+2. To check the status of the service, run:
+```bash
+sky serve status code-llama
+```
+After a while, you will see the following output:
+```console
+Services
+NAME        VERSION  UPTIME  STATUS        REPLICAS  ENDPOINT            
+code-llama  1        -       READY         2/2       3.85.107.228:30002  
+
+Service Replicas
+SERVICE_NAME  ID  VERSION  IP  LAUNCHED    RESOURCES                   STATUS REGION  
+code-llama    1   1        -   2 mins ago  1x Azure({'A100-80GB': 2})) READY  eastus  
+code-llama    2   1        -   2 mins ago  1x GCP({'L4': 8}))          READY  us-east4-a 
+```
+As shown, the service is now backed by 2 replicas, one on Azure and one on GCP, and the accelerator
+type is chosen to be **the cheapest and available one** on the clouds. That said, it maximizes the
+availability of the service while minimizing the cost.
+
+3. To access the model, we use the same curl command to send the request to the endpoint:
+```bash
+ENDPOINT=$(sky serve status --endpoint code-llama)
+
+curl -L http://$ENDPOINT/v1/completions \
+    -H "Content-Type: application/json" \
+    -d '{
+      "model": "codellama/CodeLlama-70b-Python-hf",
+      "prompt": "def quick_sort(a: List[int]):",
+      "max_tokens": 256
+  }' | jq -r '.choices[0].text'
+```
+
+## **Optional:** Chat with Code Llama
+
+We can also chat with Code Llama using the same endpoint by using the `codellama/CodeLlama-70b-Instruct-hf` model.
+To test it, we first replace the model in the [endpoint.yaml](endpoint.yaml) with the instruction-tuned model:
+```bash
+sed -i 's/codellama\/CodeLlama-70b-Python-hf/codellama\/CodeLlama-70b-Instruct-hf/g' endpoint.yaml
+```
+Then, we launch the service with the new endpoint:
+```bash
+sky serve up -n code-llama ./endpoint.yaml
+```
+Or, we can update the existing service:
+```bash
+sky serve update code-llama ./endpoint.yaml
+```
+
+```bash
+ENDPOINT=$(sky serve status --endpoint code-llama-instruct)
+
+curl -L http://$ENDPOINT/v1/chat/completions \
```

Comment:
Wondering - is there some fastchat like web interface that we could use here? Having a in-browser chat interface with code llama would be a powerful demo!

Reply:
Great point! Just added a web GUI that connects to the endpoint our skyserve starts, and a demo GIF in the readme. : )